src/stp/new_plays/Pass.cpp

```diff
@@ -0,0 +1,98 @@
+namespace rtt::ai::stp {
+
+Pass::Pass() {
+    roles = std::array<std::unique_ptr<Role>, rtt::ai::Constants::ROBOT_COUNT()>{
+                std::make_unique<Passer>(Passer("passer")), std::make_unique<PassReceiver>(PassReceiver("pass_receiver")), std::make_unique<TestRole>(TestRole("defender1")),
+                std::make_unique<TestRole>(TestRole("test_role_3")), std::make_unique<TestRole>(TestRole("test_role_4")), std::make_unique<TestRole>(TestRole("test_role_5")),
+                std::make_unique<TestRole>(TestRole("test_role_6")), std::make_unique<TestRole>(TestRole("test_role_7")), std::make_unique<TestRole>(TestRole("test_role_8")),
+                std::make_unique<TestRole>(TestRole("test_role_9")), std::make_unique<TestRole>(TestRole("test_role_10"))};
+}
+
+bool Pass::isValidPlay(world_new::World* world) noexcept { return true; }
+
+uint8_t Pass::score(world_new::World* world) noexcept { return 13; }
+
+void Pass::assignRoles() noexcept {
+    Dealer dealer{world->getWorld().value(), &field};
+
+    Dealer::FlagMap flagMap;
+    Dealer::DealerFlag closeToBallFlag(DealerFlagTitle::CLOSE_TO_BALL, DealerFlagPriority::HIGH_PRIORITY);
+    Dealer::DealerFlag closeToTheirGoalFlag(DealerFlagTitle::CLOSE_TO_THEIR_GOAL, DealerFlagPriority::MEDIUM_PRIORITY);
+    Dealer::DealerFlag notImportant(DealerFlagTitle::CLOSE_TO_OUR_GOAL, DealerFlagPriority::LOW_PRIORITY);
+
+    flagMap.insert({"passer", {closeToBallFlag}});
+    flagMap.insert({"pass_receiver", {closeToTheirGoalFlag}});
+    flagMap.insert({"defender1", {notImportant}});
+    flagMap.insert({"test_role_3", {closeToTheirGoalFlag}});
+    flagMap.insert({"test_role_4", {closeToBallFlag}});
+    flagMap.insert({"test_role_5", {closeToTheirGoalFlag, closeToBallFlag}});
+    flagMap.insert({"test_role_6", {closeToBallFlag}});
+    flagMap.insert({"test_role_7", {closeToTheirGoalFlag}});
+    flagMap.insert({"test_role_8", {closeToTheirGoalFlag, closeToBallFlag}});
+    flagMap.insert({"test_role_9", {closeToBallFlag}});
+    flagMap.insert({"test_role_10", {closeToTheirGoalFlag}});
+
+    auto distribution = dealer.distribute(world->getWorld()->getUs(), flagMap);
+
+    stpInfos = std::unordered_map<std::string, StpInfo>{};
+    for (auto& role : roles) {
+        auto roleName{role->getName()};
+        if (distribution.find(roleName) != distribution.end()) {
+            auto robot = distribution.find(role->getName())->second;
+
+            stpInfos.emplace(roleName, StpInfo{});
+            stpInfos[roleName].setRobot(robot);
+        }
+    }
+}
+
+void Pass::calculateInfoForRole() noexcept {
+    // Calculate most important positions to defend
+    // You know you have n defenders, because the play assigned it that way
+    auto enemyRobots = world->getWorld()->getThem();
+    auto defensivePositions = calculateDefensivePositions(2, world, enemyRobots);
+
+    // Use this new information to assign the roles using the dealer.
+    decltype(stpInfos)::iterator stpInfos_iter{};
```

Comment:
Uhm, I guess this is an option, but can't we do it somehow else? I'm not a very big fan o huge switch statements, but maybe that works better for now

Reply:
Yeah we need to think about the best way to do this, there is probably a better way. I will add a TODO.

Reply:
Do what? The dealer assignment?

Reply:
The assignment of role-specific data